packages/utils/src/composeAbortSignals.ts

```diff
@@ -10,29 +10,52 @@ export type ComposedAbortSignal = AbortSignal & { destroy: () => void }
  * aforementioned instance of AbortSignal will have more and more listeners added
  * but never cleaned.
  */
-export function composeAbortSignals(...signals: AbortSignal[]): ComposedAbortSignal {
-    if (signals.length === 0) {
-        throw new Error('must provide at least one AbortSignal')
-    }
+export function composeAbortSignals(
+    ...signals: (AbortSignal | undefined | null)[]
+): ComposedAbortSignal {
+    const abortController = new AbortController()
+
+    for (const signal of signals) {
+        if (signal?.aborted) {
+            abortController.abort()
 
-    const preAbortedSignal = signals.find((s) => s.aborted)
-    if (preAbortedSignal !== undefined) {
-        return Object.assign(preAbortedSignal, { destroy: () => {} })
+            return Object.assign(abortController.signal, { destroy: () => {} })
```

Comment:
@harbu there u have it. I've used the internal signal as \*the* composed signal. No point in assigning `destroy` to one of the given signals (that's a no-no).